feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapMainSettingsViewModel.kt

```diff
@@ -163,30 +166,39 @@ class SwapMainSettingsViewModel(
         emptyAssetTitle = R.string.swap_field_asset_from_title,
         maxActionProvider = assetInFlow
             .providingMaxOf(Asset::transferableInPlanks)
-            .deductFee(feeMixin, SwapFee::totalDeductedPlanks)
+            .deductFee(feeMixin, SwapFee::totalDeductedPlanks),
+        fieldValidator = getAmountInFieldValidator()
     )
 
     val amountOutInput = swapAmountInputMixinFactory.create(
         coroutineScope = viewModelScope,
         tokenFlow = assetOutFlow.token().nullOnStart(),
         emptyAssetTitle = R.string.swap_field_asset_to_title,
-        fiatFormatter = swapInputMixinPriceImpactFiatFormatterFactory.create(priceImpact)
+        fiatFormatter = swapInputMixinPriceImpactFiatFormatterFactory.create(priceImpact),
+        fieldValidator = getAmountOutFieldValidator()
     )
 
     val rateDetails: Flow<ExtendedLoadingState<String>> = quotingState.map {
         when (it) {
-            is QuotingState.NotAvailable, QuotingState.Loading -> ExtendedLoadingState.Loading
+            is QuotingState.NotAvailable, QuotingState.Loading, QuotingState.Default -> ExtendedLoadingState.Loading
             is QuotingState.Loaded -> ExtendedLoadingState.Loaded(formatRate(it.value))
         }
     }
         .shareInBackground()
 
-    val showDetails: Flow<Boolean> = quotingState.map { it !is QuotingState.NotAvailable }
+    val showDetails: Flow<Boolean> = quotingState.map { it !is QuotingState.NotAvailable && it !is QuotingState.Default }
         .shareInBackground()
 
     private val _validationProgress = MutableStateFlow(false)
 
-    val buttonState: Flow<DescriptiveButtonState> = flowOf { DescriptiveButtonState.Enabled(resourceManager.getString(R.string.common_continue)) }
+    val buttonState: Flow<DescriptiveButtonState> = combine(
+        accumulate(amountInInput.fieldError, amountOutInput.fieldError),
```

Comment:
Just a side note  on `accumulate`. Here is our current implementation:
![image](https://github.com/novasamatech/nova-wallet-android/assets/70131744/7fb33486-8399-4e87-bc0a-2d4e6cf9955e)
I've recently realized we can utilize `runningFold` to achieve much cleaner version. For example, here is the code I've written for maps:
![image](https://github.com/novasamatech/nova-wallet-android/assets/70131744/bf4fb9f4-893b-4322-b8a1-c2592d207f4e)
The only side effect of such refactoring is that ordering of original list might not be preserved